Enyim.Caching/Memcached/MemcachedNode.cs

```diff
@@ -387,6 +394,95 @@ internal async Task InitPoolAsync()
                 }
             }
 
+            private void StartReconciliationTask()
+            {
+                if (_connectionIdleTimeout == TimeSpan.Zero)
+                    return;
+
+                var reconcileTimer = new PeriodicTimer(_connectionIdleTimeout);
+                _ = RunTimer();
+
+                async Task RunTimer()
+                {
+                    while (await reconcileTimer.WaitForNextTickAsync().ConfigureAwait(false))
+                    {
+                        try
+                        {
+                            using var source = new CancellationTokenSource(_connectionIdleTimeout);
+                            await reconcileAsync(source.Token).ConfigureAwait(false);
+                        }
+                        catch(Exception e)
+                        {
+                            _logger.LogWarning("ReconciliationTaskFailed", new EventId(0), e);
+                        }
+                    }
+                }
+
+            }
+
+            private async Task reconcileAsync(CancellationToken cancellationToken)
+            {
+                // synchronize access to this method as only one clean routine should be run at a time
+                await _cleanSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
+                try
+                {
+                    var waitTimeout = TimeSpan.FromMilliseconds(10);
+                    while (true)
+                    {
+                        lock (_freeItems)
+                        {
+                            if (maxItems - _semaphore.CurrentCount + _freeItems.Count <= minItems)
+                                return;
+                        }
+
+
+                        if (!await _semaphore.WaitAsync(waitTimeout, cancellationToken).ConfigureAwait(false))
+                            return;
+
+                        try
+                        {
+                            PooledSocket retval = null;
+                            lock (_freeItems)
+                            {
+                                if (_freeItems.Count > 0)
+                                {
+                                    retval = _freeItems.Last!.Value;

```

Comment:
Just a side question - What happens when you have added an exclamation mark and the object turns out to be null?

Reply:
I've added check in line 455. (  there will always be more than minPoolSize items in the list )